store/models/service_agreement.go

```diff
@@ -63,10 +66,10 @@ func BuildServiceAgreement(us UnsignedServiceAgreement, signer Signer) (ServiceA
 	}
 
 	jobSpec := NewJob()
-	jobSpec.Initiators = us.ServiceAgreementRequest.Initiators
-	jobSpec.Tasks = us.ServiceAgreementRequest.Tasks
-	jobSpec.EndAt = us.ServiceAgreementRequest.EndAt
-	jobSpec.StartAt = us.ServiceAgreementRequest.StartAt
+	jobSpec.Initiators = us.JobSpecRequest.Initiators
+	jobSpec.Tasks = us.JobSpecRequest.Tasks
```

Comment:
Optional: Because this is so removed from the `NewJob` factory method, we could easily overlook updating the assignments here should `jobspec` be updated. Let's create a `NewJobFromRequest(us.JobSpecRequest)` to dry up some code in tests and keep constructors like this close together so if we update one we know to update the other.